### PR TITLE
docs: Update registry index file and main readme

### PR DESCRIPTION
**Pull request by marclop**
## Description

Updates the main index file to follow a different structure.

I've removed the example usage from the Terraform Registry index since
those are covered in detail in each of the resources.

## How Has This Been Tested?

https://registry.terraform.io/tools/doc-preview

## Types of Changes

- [x] Documentation

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -5,66 +5,52 @@ page_title: "Provider: Elastic Cloud"
...
+API keys are the recommended authentication method. They can be used to authenticate against Elasticsearch Service or Elastic Cloud Enterprise.
 
-## Authentication
+#### Generating an Elasticsearch Service (ESS) API Key
 
-The Elastic Cloud Terraform provider offers two methods of authentication against the remote API: `apikey` or a combination of `username` and `password`. Depending on the environment, one or both can be used. The Public API of Elasticsearch Service (ESS) is the default `endpoint` that the provider will target.
+To generate an API key, follow these steps:
 
-Elasticsearch Service (ESS) only supports `apikey`. Elastic Cloud Enterprise (ECE) supports both `apikey` and a combination of `username` and `password`.
+  1. Navigate to <https://cloud.elastic.co/login> with your browser
```
Comment (alaudazzi) [inline docs/index.md:34]:
```suggestion
  1. Open you browser and navigate to <https://cloud.elastic.co/login>.
```

File: docs/index.md [hunk 2]
```diff
@@ -5,66 +5,52 @@ page_title: "Provider: Elastic Cloud"
...
+  2. Log in with your Email and Password.
```
Comment (alaudazzi) [inline docs/index.md:35]:
```suggestion
  2. Log in with your email and password.
```

File: docs/index.md [hunk 3]
```diff
@@ -5,66 +5,52 @@ page_title: "Provider: Elastic Cloud"
...
+  3. Click on [Elasticsearch Service](https://cloud.elastic.co/deployments).
+  4. Navigate to [Featuress > API Keys](https://cloud.elastic.co/deployment-features/keys) and click on **Generate API Key**.
```
Comment (alaudazzi) [inline docs/index.md:37]:
```suggestion
  4. Navigate to [Features > API Keys](https://cloud.elastic.co/deployment-features/keys) and click on **Generate API Key**.
```

File: docs/index.md [hunk 4]
```diff
@@ -5,66 +5,52 @@ page_title: "Provider: Elastic Cloud"
...
+  5. Chose a name for your API key.
```
Comment (alaudazzi) [inline docs/index.md:38]:
```suggestion
  5. Choose a name for your API key.
```

File: docs/index.md [hunk 5]
```diff
@@ -5,66 +5,52 @@ page_title: "Provider: Elastic Cloud"
...
+  6. Persist your API key somewhere safe.
```
Comment (alaudazzi) [inline docs/index.md:39]:
```suggestion
  6. Save your API key somewhere.
```

File: docs/index.md [hunk 6]
```diff
@@ -5,66 +5,52 @@ page_title: "Provider: Elastic Cloud"
...
 
-!> **Warning:** Hard-coding credentials into any Terraform configuration is not
-recommended, and risks secret leakage should this file ever be committed to a
-public version control system.
+#### Using your API Key on the Elastic Cloud terraform provider
 
-### API key authentication (recommended)
+After you've generated your API Key, you can make it available to the Terraform provider by exporting it as the environment variable `EC_API_KEY` environment variable (recommended) or hardcoded in the provider `.tf` configuration file (supported but not recommended).
```
Comment (alaudazzi) [inline docs/index.md:43]:
```suggestion
After you've generated your API Key, you can make it available to the Terraform provider by exporting it as the environment variable `EC_API_KEY` (recommended), or hardcoded in the provider `.tf` configuration file (supported but not recommended).
```

**Review by alaudazzi — APPROVED** [review]
Suggested a few edits. LGTM.

**Review by karencfv — APPROVED** [review]
Thanks for working on this! 🙇‍♀️

**Pull request merged**